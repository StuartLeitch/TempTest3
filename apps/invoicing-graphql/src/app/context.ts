import {Knex, KnexInvoiceRepo, KnexTransactionRepo} from '@hindawi/shared';

export interface ReposContext {
  invoice: KnexInvoiceRepo;
  transaction: KnexTransactionRepo;
}

export interface Context {
  repos: ReposContext;
}

export type ContextCreator = (params: any) => Context;

export function makeContext(db: Knex): ContextCreator {
  return () => ({
    repos: {
      invoice: new KnexInvoiceRepo(db),
      transaction: new KnexTransactionRepo(db)
    }
  });
}
