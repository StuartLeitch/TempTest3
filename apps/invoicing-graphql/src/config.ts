import path from 'path';
import { environment } from '@env/environment';

const dbConfig = {
  dbUser: process.env.DB_USERNAME || environment.DB_USERNAME,
  dbHost: process.env.DB_HOST || environment.DB_HOST,
  dbDatabase: process.env.DB_DATABASE || environment.DB_DATABASE,
  dbPassword: process.env.DB_PASSWORD,
  dbMigrationsDir:
    process.env.DB_MIGRATIONS_DIR ||
    path.join(__dirname, environment.DB_MIGRATIONS_DIR),
  dbSeedsDir:
    process.env.DB_SEEDS_DIR || path.join(__dirname, environment.DB_SEEDS_DIR)
};

export interface SalesForceConfig {
  user: string;
  password: string;
  loginUrl: string;
  securityToken: string;
}

export interface PayPalConfig {
  clientSecret: string;
  environment: string;
  clientId: string;
}

export class Config {
  dbUser: string;
  dbHost: string;
  dbDatabase: string;
  dbPassword: string;
  dbMigrationsDir: string;
  dbSeedsDir: string;

  salesForce: SalesForceConfig;
  payPal: PayPalConfig;

  feRoot: string;

  sanctionedCountryNotificationReceiver: string;
  sanctionedCountryNotificationSender: string;

  constructor() {
    Object.assign(this, dbConfig);

    this.salesForce = {
      user: process.env.SAGE_USER,
      password: process.env.SAGE_PASSWORD,
      loginUrl: process.env.SAGE_LOGIN_URL,
      securityToken: process.env.SAGE_SECURITY_TOKEN
    };

    this.payPal = {
      clientSecret: process.env.PP_CLIENT_SECRET,
      environment: process.env.PP_ENVIRONMENT,
      clientId: process.env.PP_CLIENT_ID
    };

    this.sanctionedCountryNotificationReceiver =
      process.env.SANCTIONED_COUNTRY_NOTIFICATION_RECEIVER;
    this.sanctionedCountryNotificationSender =
      process.env.SANCTIONED_COUNTRY_NOTIFICATION_SENDER;
  }
}

export function makeConfig(): Config {
  return new Config();
}

export const config = makeConfig();
