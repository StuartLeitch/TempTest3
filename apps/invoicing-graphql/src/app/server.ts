import {ApolloServer} from 'apollo-server';

import {ContextCreator} from './context';
import {typeDefs} from './schema';
import {resolvers} from './resolvers';

export function makeServer(context: ContextCreator): ApolloServer {
  return new ApolloServer({
    typeDefs,
    resolvers,
    context
  });
}
