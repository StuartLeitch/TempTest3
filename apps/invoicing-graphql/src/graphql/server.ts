import {ApolloServer} from 'apollo-server-express';
import express from 'express';

import {ContextCreator} from './context';
import {typeDefs} from './schema';
import {resolvers} from './resolvers';

const app = express();

app.get('/api/test', (req, res) => res.status(200).send('asdf'));


export function makeServer(context: ContextCreator): ApolloServer {
  return new ApolloServer({
    typeDefs,
    resolvers,
    context
  });
}
