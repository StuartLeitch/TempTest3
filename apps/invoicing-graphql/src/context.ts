import Knex from 'knex';
import {
  KnexPaymentMethodRepo,
  KnexAddressRepo,
  KnexArticleRepo,
  KnexInvoiceItemRepo,
  KnexInvoiceRepo,
  KnexTransactionRepo,
  KnexPaymentRepo,
  KnexPayerRepo,
  KnexWaiverRepo,
  KnexCatalogRepo,
  VATService,
  WaiverService,
  EmailService
} from '@hindawi/shared';
import { Config } from './config';
import { CheckoutService } from './services/checkout';
import { AuthService } from './services/auth';
import { ExchangeRateService } from './../../../libs/shared/src/lib/domain/services/ExchangeRateService';
import { ErpService } from './services/erp';

const checkoutNodeJsSDK = require('@paypal/checkout-server-sdk');

export interface ReposContext {
  address: KnexAddressRepo;
  invoice: KnexInvoiceRepo;
  catalog: KnexCatalogRepo;
  invoiceItem: KnexInvoiceItemRepo;
  transaction: KnexTransactionRepo;
  payer: KnexPayerRepo;
  payment: KnexPaymentRepo;
  paymentMethod: KnexPaymentMethodRepo;
  waiver: KnexWaiverRepo;
  manuscript: KnexArticleRepo;
}

export interface Context {
  repos: ReposContext;
  checkoutService: CheckoutService;
  authService: AuthService;
  vatService: VATService;
  waiverService: WaiverService;
  payPalService: any;
  emailService: EmailService;
  exchangeRateService: ExchangeRateService;
  erpService: ErpService;
}

function makePayPalEnvironment(
  clientId: string,
  clientSecret: string,
  environment: string
) {
  if (environment === 'live' || environment === 'production') {
    return new checkoutNodeJsSDK.core.LiveEnvironment(clientId, clientSecret);
  } else {
    return new checkoutNodeJsSDK.core.SandboxEnvironment(
      clientId,
      clientSecret
    );
  }
}

function makePayPal(config: Config) {
  const { clientId, environment, clientSecret } = config.payPal;
  return new checkoutNodeJsSDK.core.PayPalHttpClient(
    makePayPalEnvironment(clientId, clientSecret, environment)
  );
}

export function makeContext(config: Config, db: Knex): Context {
  return {
    repos: {
      address: new KnexAddressRepo(db),
      catalog: new KnexCatalogRepo(db),
      invoice: new KnexInvoiceRepo(db),
      invoiceItem: new KnexInvoiceItemRepo(db),
      transaction: new KnexTransactionRepo(db),
      payer: new KnexPayerRepo(db),
      payment: new KnexPaymentRepo(db),
      paymentMethod: new KnexPaymentMethodRepo(db),
      waiver: new KnexWaiverRepo(db),
      manuscript: new KnexArticleRepo(db)
    },
    checkoutService: new CheckoutService(),
    authService: new AuthService(config),
    vatService: new VATService(),
    waiverService: new WaiverService(),
    payPalService: () => makePayPal(config),
    emailService: new EmailService(),
    exchangeRateService: new ExchangeRateService(),
    erpService: new ErpService(config)
  };
}
