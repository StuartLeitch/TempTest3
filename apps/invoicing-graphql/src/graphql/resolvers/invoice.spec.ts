import {createTestClient} from 'apollo-server-testing';
import gql from 'graphql-tag';
import Knex from 'knex';
import path from 'path';

import {Config} from '../../config';
import {makeServer} from '../server';
import {makeContext} from '../../context';

const migrationsDirPath = path.resolve(
  __dirname,
  '../../../../../libs/shared/src/lib/infrastructure/database/knex/migrations'
);

const config = new Config();
const db = Knex({
  client: 'sqlite3',
  pool: {
    min: 1,
    max: 1
  },
  migrations: {
    directory: migrationsDirPath
  },
  connection: ':memory:',
  useNullAsDefault: true
});

describe('Query.invoice', () => {
  let server: any;
  let context: any;
  let client: any;

  beforeAll(async () => {
    await db.migrate.latest();

    context = makeContext(config, db);
    server = makeServer(context);
    client = createTestClient(server);
  });

  afterAll(async () => db.destroy());

  it('should work', async () => {
    const query = gql`
      query invoice($id: String!) {
        invoice(id: $id) {
          id
          status
        }
      }
    `;

    await db('invoices').insert({
      id: 'invoice-1',
      transactionId: 'transaction-1',
      status: 0
    });

    const res = await client.query({
      query,
      variables: {
        id: 'invoice-1'
      }
    });
    expect(res.data).toMatchSnapshot();
  });
});
