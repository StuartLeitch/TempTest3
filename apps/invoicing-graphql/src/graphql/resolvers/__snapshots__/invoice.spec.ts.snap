// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query.invoice should work 1`] = `
Object {
  "invoice": Object {
    "id": "invoice-1",
    "status": "DRAFT",
  },
}
`;
